src/nl/rubensten/texifyidea/intentions/LatexLabelDefiningNewCommand.kt

```diff
@@ -0,0 +1,112 @@
+                ?: parentElement.parentOfType(LatexCommands::class)
+
+        // when element is \label and parent is \newcommand check if command is already in list
+        if (selected.name == "\\label" && parent?.name == "\\newcommand") {
+            // get name of the defined command
+            val cmdName = parent.requiredParameter(0) ?: return false
+            // check if there is already an entry wih this name
+            return !settings.labelCommands.containsKey(cmdName)
+        }
+
+        // check if the parent of the current position is \newcommand, if true, set the selected command to the parent
+        // command
+        if (parent?.name == "\\newcommand") {
+            selected = parent
+        }
+
+        // when command is \newcommand, check if it contains a \label
+        if (selected.name == "\\newcommand") {
+            val children = selected.childrenOfType(LatexCommands::class)
+            if (children.none { it.name == "\\label" }) {
+                return false
+            }
+            val cmdName = selected.requiredParameter(0) ?: return false
+            return !settings.labelCommands.containsKey(cmdName)
+        }
+
+        return false
+    }
+
+    override fun invoke(project: Project, editor: Editor?, file: PsiFile?) {
+        if (editor == null || file == null || !file.isLatexFile()) {
+            return
+        }
+
+        // check if selected part is \label
+        val element = file.findElementAt(editor.caretModel.offset) ?: return
+        val selected = element as? LatexCommands ?: element.parentOfType(LatexCommands::class) ?: return
+
+        // get parent element
+        val parentElement = selected.parent
+        val parent = parentElement as? LatexCommands
+                ?: parentElement.parentOfType(LatexCommands::class)
+
+        val label: LatexCommands
+        val newCommand: LatexCommands
+
+        // map correct values to label and newCommand
+        if (selected.name == "\\label" && parent?.name == "\\newcommand") {
+            label = selected
+            newCommand = parent
+        }
+        else if (selected.name == "\\newcommand") {
+            label = selected.childrenOfType(LatexCommands::class)
+                    .firstOrNull { it.name == "\\label" } ?: return
+            newCommand = selected
+        }
+        else if (parent?.name == "\\newcommand") {
+            label = parent.childrenOfType(LatexCommands::class)
+                    .firstOrNull { it.name == "\\label" } ?: return
+            newCommand = parent
+        }
+        else {
+            return
+        }
+
+        // get the name of the command
+        val commandName = newCommand.requiredParameter(0) ?: return
+        // get the position of the label in new command, identified by the argument number of the first required parameter
+        // of label command
+        val position = label.requiredParameter(0)?.replace("#", "")
+                ?.toIntOrNull() ?: 1
```

Comment:
Either split it up (each statement on a new line), or put it on one line.

Reply:
fixed in commit e43885c396a6aec2c17f9d63fb39fa2ae1c7c550